### PR TITLE
Update Tetris

### PR DESCRIPTION
**Pull request by YCF**
Add a fake row to config the encoder's keycode easily.
Auto change the rgblight color with the keymap.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
>Add a fake row to config the encoder's keycode easily.

Any reason you're not using the normal encoder feature?

**Comment by YCF** [comment]
It's using the normal encoder feature,I just make it can be configure the keycode via the configure.qmk.fm.
and also,some RGB light with the encoder.

File: keyboards/tetris/config.h [hunk 1]
```diff
@@ -39,19 +34,21 @@
...
 #endif
 
-#if RGBLIGHT_ENABLE
+#define NUMBER_OF_ENCODERS 2
+#define ENCODERS_PAD_A { D1,F1 }
+#define ENCODERS_PAD_B { D0,F0 }
+
 #define RGB_DI_PIN F5
 #define RGBLIGHT_ANIMATIONS
-#define RGBLIGHT_SLEEP
```
Comment (drashna) [inline keyboards/tetris/config.h]: No more rgb sleep?

File: keyboards/tetris/readme.md [hunk 2]
```diff
@@ -2,6 +2,8 @@ Tetris
 ===
 A compact 40% (12x4) ortholinear keyboard (Planck Mit layout) with per switch RGB and encoder made by Fengz. 
 
+Fake keys at the 4th row is to config the encoder's keycode.
```
Comment (drashna) [inline keyboards/tetris/readme.md]: @yanfali .... maybe we need configurator support for encoders? :)

**Comment by yanfali** [comment]
Needs API support. Config.h changes

File: keyboards/tetris/readme.md (hunk 2 above)
Comment (noroadsleft) [inline keyboards/tetris/readme.md]: The fake keys are actually on the fifth row. Grammar fixes here as well.

```suggestion
Fake keys on the 5th row are for configuring the encoder's keycode.
```

File: keyboards/tetris/readme.md [hunk 3]
```diff
@@ -12,4 +14,4 @@ Make example for this keyboard (after setting up your build environment):
 
     make tetris:default
 
-See [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) then the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information.
\ No newline at end of file
+See [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) then the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information.
```
Comment (noroadsleft) [inline keyboards/tetris/readme.md]: Could you update this please?

```suggestion
See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

**Comment by drashna** [comment]
@YCF I'm mostly curious, but why did you remove the rgb light sleep functionality?  Issues with it, or cosmetic choice, or?

**Comment by YCF** [comment]
@drashna oh,my mistake.please help me add it.
I can use my personal pc at the weekend only,thanks you.

File: keyboards/tetris/config.h (hunk 1 above)
Comment (drashna) [reply keyboards/tetris/config.h]:
```suggestion
#define RGBLIGHT_SLEEP
```

File: keyboards/tetris/config.h [hunk 4]
```diff
@@ -39,19 +34,21 @@
...
   #define NO_MUSIC_MODE
 #endif
 
-#if RGBLIGHT_ENABLE
+#define NUMBER_OF_ENCODERS 2
+#define ENCODERS_PAD_A { D1,F1 }
+#define ENCODERS_PAD_B { D0,F0 }
+
 #define RGB_DI_PIN F5
 #define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/tetris/config.h]:
```suggestion
#define RGBLIGHT_ANIMATIONS
#define RGBLIGHT_SLEEP
```

**Comment by YCF** [comment]
Hopefully this update will allow Encoder to work on QMK config.

**Comment by YCF** [comment]
@drashna

**Comment by drashna** [comment]
Sorry!  Thanks!

**Pull request merged**